src/materials/ConcreteThermalMoisture.C

```diff
@@ -160,125 +134,74 @@ ConcreteThermalMoisture::ConcreteThermalMoisture(const InputParameters & paramet
-                               : 1100.0;
-    _input_thermal_conductivity = isParamValid("ref_thermal_conductivity_of_concrete")
-                                      ? getParam<Real>("ref_thermal_conductivity_of_concrete")
-                                      : 3.0;
-    _aggregate_type = isParamValid("aggregate_type") ? getParam<MooseEnum>("aggregate_type") : 0;
-    parameter_error_check = false;
+    if (!parameters.isParamSetByUser("ref_density"))
+      mooseError("For CONSTANT thermal_model, ref_density must be defined.");
+    if (!parameters.isParamSetByUser("ref_specific_heat"))
+      mooseError("For constant thermal transport model, ref_specific_heat must be defined.");
+    if (!parameters.isParamSetByUser("ref_thermal_conductivity"))
+      mooseError("For constant thermal transport model, ref_thermal_conductivity must be defined.");
   }
-  if (parameter_error_check)
+  else if (_thermal_model == "ASCE-1992")
   {
-    if (_thermal_model == "CONSTANT")
-    {
-      if (!parameters.isParamSetByUser("ref_density"))
-        mooseError("For CONSTANT thermal_model, ref_density must be defined.");
-      if (!parameters.isParamSetByUser("ref_specific_heat"))
-        mooseError("For constant thermal transport model, ref_specific_heat must be defined.");
-      if (!parameters.isParamSetByUser("ref_thermal_conductivity"))
-        mooseError(
-            "For constant thermal transport model, ref_thermal_conductivity must be defined.");
-    }
-    else if (_thermal_model == "ASCE-1992")
-    {
-      if (!parameters.isParamSetByUser("aggregate_type"))
-        mooseError("For ASCE-1992 thermal model, aggregate_type must be defined.");
-    }
-    else if (_thermal_model == "KODUR-2004")
-    {
-      if (!parameters.isParamSetByUser("aggregate_type"))
-        mooseError("For KODUR-2004 thermal model, aggregate_type must be defined.");
-    }
-    else if (_thermal_model == "EUROCODE-2004")
-    {
-      if (!parameters.isParamSetByUser("ref_density"))
-        mooseError("For EUROCODE-2004 thermal model, ref_density must be defined.");
-      if (!parameters.isParamSetByUser("ref_specific_heat"))
-        mooseError("For EUROCODE-2004 thermal model, ref_specific_heat must be defined.");
-    }
+    if (!parameters.isParamSetByUser("aggregate_type"))
+      mooseError("For ASCE-1992 thermal model, aggregate_type must be defined.");
   }
-
-  parameter_error_check = true;
-
-  if (parameters.isParamSetByUser("moisture_diffusivity_model"))
+  else if (_thermal_model == "KODUR-2004")
   {
-    _moisture_model = getParam<MooseEnum>("moisture_diffusivity_model");
-    _cement_mass = isParamValid("cement_mass") ? getParam<Real>("cement_mass") : 354.0;
-    _water_to_cement =
-        isParamValid("water_to_cement_ratio") ? getParam<Real>("water_to_cement_ratio") : 0.43;
-    _cement_type = isParamValid("cement_type") ? getParam<MooseEnum>("cement_type") : 0;
```

Comment:
Do you still need to set or check all of these defaults for the moisture_diffusivity_models?

Reply:
Good point. The way the new syntax (which we've been using for a while) works, those are all effectively required parameters for whatever model you're using. We should at least capture these recommended values, maybe just as comments in the code, so we don't lose them.